sdk/core/Azure.Core/src/DynamicData/MutableJsonElement.cs

```diff
@@ -239,6 +244,7 @@ private static string GetFormatExceptionText(string path, Type type)
         ///   <see langword="false"/> otherwise.
         /// </returns>
         /// <exception cref="InvalidOperationException">This value's <see cref="ValueKind"/> is not <see cref="JsonValueKind.Number"/>.</exception>
+        [RequiresUnreferencedCode(SerializationRequiresUnreferencedCode)]
```

Comment:
I wonder if it would be simpler to do the following:
* Annotate the constructor of the struct with RequiresUnreferencedCode (RUC) - as it seems that there's barely an operation which can be done on it without it being RUC anyway (it seems several methods here are missing the annotations - for example `GetInt64` should have RUC since it calls `TryGetInt64`.
* Add a type-level `UnconditionalSuppressedMessage("", "IL2026", ...)` - so that all members don't need RUC or suppressions.

This should have two effects:
* Simplify the annotations - only 2 attributes added
* Lower the amount of warnings produced by code using this class as only the places which create a new instance would get a warning.

The overall result would be similar to what would happen if the RUC could be placed on the type itself (currently impossible as it's a struct, on a class that would be the behavior)

This working correctly makes several assumptions, but they're relatively easy to maintain:
* The compiler generated .ctor is never actually used - given that pretty much everything on the class will throw NRE if that happens, I think it's a safe assumption
* Static methods don't have trimming issues - there are only 3 statics which are simple. One way to validate this would be to add a unit test which (using reflection) checks that there are only those 3 statics on the type. The intent of the unit test would be to cause a failure if somebody adds a new method and tell them what to check for.
* There are no new nested types with trimming problems - same story as the statics - easy to make sure

@MichalStrehovsky @agocke for opinions - for context, this would be a perfect candidate for RUC on type, but it's a struct, so that's not currently possible.